pkg/api/auth/auth_mode_handler.go

```diff
@@ -38,7 +38,3 @@ func authModes() []harvesterv1.AuthenticationMode {
 func enableKubernetesCredentials() bool {
```

Comment:
Let's remove the KubernetesCredentials mode as well. So we don't need to provide the authmode option. It will be either disabled or using the Rancher auth.

Reply:
i removed the KubernetesCredentials mode as well now. There is some test failing and i'm not sure why yet or if it is related to my changes. Maybe you can give the PR another look in the meanwhile. I still have to remove the AuthenticationMode type completely but i kept it for now so that the current UI still works without any changes.

Reply:
There are missing parts that need to be removed:
```
/pkg/api/user
/pkg/apis/harvesterhci.io/v1beta1/auth.go
related crd creation in /pkg/data/crd.go
/pkg/userpreferences
user_apis_test.go and set SKIP_AUTHENTICATION for integration tests
pkg/controller/global/auth
```

Also, do a rebase to fix the CI.

If you still get test failures and have no clues after the above are addressed, feel free to let me know.

Reply:
Ok after rebasing and addressing the things you mentioned the test work thanks. Let me know i there is anything else.

Reply:
i was looking at the front end part again and noticed it queries the userpreferences. Now that they are deleted is there a replacement for that?

Reply:
I think It's using Rancher userpreferences at the moment: https://github.com/harvester/harvester/blob/master/pkg/server/router.go#L92

Reply:
looks like you are right the requests were failing for some other reason now on a fresh setup it works 👍🏻